backport_release.md

```diff
@@ -0,0 +1,84 @@
+# Backport Release Process
+
+In situations where we need to backport commits to earlier versions of our software, we'll need to perform the release process slightly differently than a normal release. This document outlines the differences between a normal release and a backport release. It uses the same outline as the [Release Guide](release.md) and will note any change between the two types of releases.
+
+## 0. Pre-Release Checklist
+
+- Get agreement on the version number to use for the backport release. This may be an intermediate number between two preexisting releases--for example a new `0.11.2` to be added between existing `0.11.1` and `0.12.0` releases. It can also be a new latest release--so `0.13.0` in the same situation--using only some of the commits that are present in the Future Release section of the release notes.
+- Get agreement on the latest commit to use for targeting the release. A backport release will be targeted on some commit other than the latest on main. Many times the new target will be an old release, which will have a tag that can be referenced--for example `v0.11.1`.
+- Get agreement on the commits to port over for the backport release.
+
+## 0.5. Port over content for backport release
+
+#### Checkout intended target commit
+
+1. Checkout the agreed upon latest commit for targeting the release. If this is a previous release, you may checkout its tag with `git checkout v0.11.1`.
+
+#### Create backport branch
+
+1. Branch off of the target commit. For the branch name, please use the most recent major and minor versions to this commit, leaving the patch number as an `x`. This means that we would create `0.11.x`. This is necessary so that if any further backport releases are needed, we could continue to use this branch as the target. This branch is to be treated as `main` is treated in a normal release. It will be the target for our release.
+2. Protect this branch to avoid unintended commits from making their way into the release undetected.
```

Comment:
I think we can add a protection rule that will automatically protect branches with `.x`

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule#about-branch-protection-rules

Reply:
I can do this, but why do we need to protect the branch?

Reply:
I think there are two reasons to protect the branch
- Avoid accidentally deleting the branch. If we end up needing to do a second backport or something, we need to have the branch around
- Make sure we get eyes on any commits that are about to be released. This was something that came up when we were doing the backport releases--it felt like it'd be very easy to accidentally include unexpected changes while cherry-picking commits. Plus there might be some further changes needed, and it'd be better to get those reviewed in their own PR than saving it for the release PR.

Reply:
I see. I am fine with this new protection rule.
One minor thing for the first point, you can go to any PR and restore the branch.